### PR TITLE
Signup testing

### PR DESCRIPTION
**Pull request by gabybrow**
## Summary
- Signup endpoint tests

## Trello Card
https://trello.com/c/KSFYVYuL/33-tests-y-documentaci%C3%B3n-de-sign-up

### CODE REVIEW

**Comment by gabybrow** [comment]
Travis is rejecting the pull request because it is not finding the database url environment variable

File: test/users.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      .post('/users')
+      .send({
+        firstName: 'xxxxxx', lastName: 'zzzzzzzzz',
+        password: '1234', email: '123456@wolox.com.ar'
+      })
+      .catch((err) => {
+        err.response.should.be.json;
+        err.response.body.should.have.property('error');
+        err.should.have.status(500);
+        // dictum.chai(err);
```
Comment (mpizzagalli) [inline test/users.js]: Remove all this comments

File: app/models/scripts/dataCreation.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
+const orm = require('../../orm');
 
 exports.execute = (db) => {
-
+  // orm.models.user.create({ firstName: 'test', lastName: 'wolox', password: '123456789', email: 'test@wolox.com.ar' }).then();
```
Comment (mpizzagalli) [inline app/models/scripts/dataCreation.js]: We should try figure out why this isn't working right, so we can add it in this PR.

**Review by mpizzagalli — CHANGES_REQUESTED** [review]

File: test/users.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+const chai = require('chai'),
+  dictum = require('dictum.js'),
+  server = require('./../app'),
+  orm = require('../app/orm'),
```
Comment (mpizzagalli) [inline test/users.js]: you don't need this require anymore.

File: test/users.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  it('should fail because the password is too short', (done) => {
+    chai.request(server)
+      .post('/users')
+      .send({
+        firstName: 'xxxxxx', lastName: 'zzzzzzzzz',
+        password: '1234', email: '123456@wolox.com.ar'
+      })
+      .catch((err) => {
+        err.response.should.be.json;
+        err.response.body.should.have.property('error');
```
Comment (mpizzagalli) [inline test/users.js]: You could also check that the error message is the one we expect in this case `The password is too short`

File: test/users.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  it('should fail because something is missing', (done) => {
+    chai.request(server)
+      .post('/users')
+      .send({
+        firstName: 'xxxxxx',
+        password: '1234', email: '123456@wolox.com.ar'
+      })
+      .catch((err) => {
+        err.response.should.be.json;
+        err.response.body.should.have.property('error');
```
Comment (mpizzagalli) [inline test/users.js]: same here

File: test/users.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  it('should fail because of existing email', (done) => {
+    chai.request(server)
+      .post('/users')
+      .send({
+        firstName: 'xxxxxx', lastName: 'yyyyyyy',
+        password: '123456789', email: 'test@wolox.com.ar'
+      })
+      .catch((err) => {
+        err.response.should.be.json;
+        err.response.body.should.have.property('error');
```
Comment (mpizzagalli) [inline test/users.js]: same here

File: test/users.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  });
+  it('should be successful', (done) => {
+    chai.request(server)
+      .post('/users')
+      .send({
+        firstName: 'aaaaa', lastName: 'bbbbb',
+        password: '123456789', email: '12345@wolox.com.ar'
+      })
+      .then((res) => {
+        res.should.have.status(200);
```
Comment (mpizzagalli) [inline test/users.js]: Here we could also check that we actually have created a new user, but for now its ok without checking.

**Review by mpizzagalli — CHANGES_REQUESTED** [review]

**Review by mpizzagalli — APPROVED** [review]

**Pull request merged**